From 084ff0722e4f3878bae698a1cad6574a2af32a1d Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Mon, 12 Mar 2012 21:43:37 +0100
Subject: [PATCH 1/3] led: simplify a bit the LED framework

Shuffle a bit the definitions in status_led.h so that it is possible
to use the LED command in common/cmd_led.c without using the status
LED stuff that automatically sets a blinking LED at boot time.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 include/status_led.h |   16 ++++++++--------
 1 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/include/status_led.h b/include/status_led.h
index c85c206..46d37ac 100644
--- a/include/status_led.h
+++ b/include/status_led.h
@@ -33,6 +33,13 @@
 #ifndef _STATUS_LED_H_
 #define	_STATUS_LED_H_
 
+/* led_id_t is unsigned long mask */
+typedef unsigned long led_id_t;
+
+extern void __led_toggle (led_id_t mask);
+extern void __led_init (led_id_t mask, int state);
+extern void __led_set (led_id_t mask, int state);
+
 #ifdef CONFIG_STATUS_LED
 
 #define STATUS_LED_OFF		0
@@ -333,14 +340,7 @@ void status_led_set  (int led, int state);
 
 #define STATUS_LED_BOOT		0	/* LED 0 used for boot status */
 
-#elif defined(CONFIG_BOARD_SPECIFIC_LED)
-/* led_id_t is unsigned long mask */
-typedef unsigned long led_id_t;
-
-extern void __led_toggle (led_id_t mask);
-extern void __led_init (led_id_t mask, int state);
-extern void __led_set (led_id_t mask, int state);
-#else
+#elif !defined(CONFIG_BOARD_SPECIFIC_LED)
 # error Status LED configuration missing
 #endif
 /************************************************************************/
-- 
1.7.4.1

